src/processor/Package.ts

```diff
@@ -424,14 +424,23 @@ export class Package {
     });
   }
 
-  private simplifyFHIROnlyRules(): void {
+  private simplifyOnlyRules(processor: FHIRProcessor): void {
     [...this.profiles, ...this.extensions].forEach(sd => {
       sd.rules.forEach(rule => {
         if (rule instanceof ExportableOnlyRule) {
           rule.types.forEach(onlyRuleType => {
-            const fhirMatch = onlyRuleType.type.match(FHIR_BASE_URL);
-            if (fhirMatch?.[1]) {
-              onlyRuleType.type = fhirMatch[1];
+            // The type might be another SD in the processor or a core FHIR resource
+            const typeSd = processor.structureDefinitions.find(sd => sd.url === onlyRuleType.type);
+            if (typeSd?.name) {
+              onlyRuleType.type = typeSd.name;
+            } else {
+              const fhirMatch = onlyRuleType.type.match(FHIR_BASE_URL);
+              if (fhirMatch?.[1]) {
+                // Only change the FHIR url into a name if it won't collide with a local SD
+                if (!processor.structureDefinitions.some(sd => sd.name === fhirMatch[1])) {
```

Comment:
I'll sing [the same song you sang](https://github.com/FHIR/sushi/pull/645#discussion_r508817143) to @julianxcarter (and I missed in my own review of his).  The above two `if` statements could be combined as a single `if` statement.  See, I learned from you, @mint-thompson!  ;-)

Reply:
hoist by my own petard!!! updating now